### PR TITLE
write AnyType

### PR DESCRIPTION
**Pull request by ziolushka**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    /// \brief getValue
+    /// \return
+    ///Extracts a value should provide type-safe access. If the requested type does not match to stored type exception is thrown
+    T getValue() {
+        char current_type = static_cast<char>(getType()[0]);
+
+        if(current_type == *typeid(T).name())
+            switch(type){
+                case Type::Int:
+                    return static_cast<int>(value.as_int);
```
Comment (Stepbura) [inline main.cpp]: conversion from 'int' to 'T', possible loss of data

File: main.cpp [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+                case Type::Float:
+                    return static_cast<float>(value.as_float);
```
Comment (Stepbura) [inline main.cpp]: conversion from 'float' to 'T', possible loss of data

File: main.cpp [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+                case Type::Double:
+                    return static_cast<double>(value.as_double);
```
Comment (Stepbura) [inline main.cpp]: possible loss of data

File: main.cpp [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+                case Type::Bool:
+                    return static_cast<bool>(value.as_bool);
```
Comment (Stepbura) [inline main.cpp]: possible loss of data

File: main.cpp [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+                case Type::Char:
+                    return static_cast<char>(value.as_char);
```
Comment (Stepbura) [inline main.cpp]: possible loss of data

**Review by Stepbura — APPROVED** [review]

File: main.cpp (hunk 1 above)
Comment (ziolushka) [reply main.cpp]: Resolved

File: main.cpp (hunk 4 above)
Comment (ziolushka) [reply main.cpp]: Resolved

File: main.cpp (hunk 5 above)
Comment (ziolushka) [reply main.cpp]: Resolved

File: main.cpp [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+
+        AnyType temp = *this;
+        *this = atObj;
+        atObj = temp;
+}
+    ///
+    /// \brief getType
+    /// \return
+    ///Returns the type of the contained value
+    string getType(){
```
Comment (Stepbura) [inline main.cpp:118]: please use const method here.

File: main.cpp [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+                    return value.as_float;
+                case Type::Double:
+                    return value.as_double;
+                case Type::Bool:
+                    return value.as_bool;
+                case Type::Char:
+                    return value.as_char;
+            }
+        else
+            throw "Bad_cast caught: Wrong type of stored value!";
```
Comment (Stepbura) [inline main.cpp:157]: It is better to use some std exception instead of string

File: main.cpp (hunk 6 above)
Comment (ziolushka) [reply main.cpp:118]: Ok. I'll fix this.

**Pull request merged**